packages/react-cosmos/src/plugins/httpProxy.ts

```diff
@@ -1,22 +1,31 @@
-import { CosmosConfig } from '../config/shared';
-import httpProxyMiddleware from 'http-proxy-middleware';
-import { DevServerPluginArgs } from '../shared/devServer';
-
-// TODO: Support advanced configuration
-// See https://github.com/react-cosmos/react-cosmos/pull/875 for context
-// Supporting all httpProxy options isn't as easy anymore, because the Cosmos
-// config is now JSON. We can either support a serializable subset of useful
-// options (like "pathRewrite") or optionally support an external config module.
-type HttpProxyConfig = { [context: string]: string };
-
-export function httpProxy({ cosmosConfig, expressApp }: DevServerPluginArgs) {
-  const httpProxyConfig = getHttpProxyCosmosConfig(cosmosConfig);
-  Object.keys(httpProxyConfig).forEach(context => {
-    const target = httpProxyConfig[context];
-    expressApp.use(context, httpProxyMiddleware(context, { target }));
-  });
-}
-
-function getHttpProxyCosmosConfig(cosmosConfig: CosmosConfig) {
-  return (cosmosConfig.httpProxy || {}) as HttpProxyConfig;
-}
+import { CosmosConfig } from '../config/shared';
+import httpProxyMiddleware from 'http-proxy-middleware';
+import { DevServerPluginArgs } from '../shared/devServer';
+
+type HttpProxyConfig = {
+  [context: string]:
+    | string
+    | {
+        target: string;
+        secure?: boolean;
+        pathRewrite?: { [rewrite: string]: string };
+        logLevel?: 'error' | 'debug' | 'info' | 'warn' | 'silent';
+      };
+};
+
+export function httpProxy({ cosmosConfig, expressApp }: DevServerPluginArgs) {
+  const httpProxyConfig = getHttpProxyCosmosConfig(cosmosConfig);
+  Object.keys(httpProxyConfig).forEach(context => {
+    let proxy = {};
+    if (typeof httpProxyConfig[context] === 'string') {
+      proxy = { target: httpProxyConfig[context] };
+    } else if (typeof httpProxyConfig[context] === 'object') {
+      proxy = httpProxyConfig[context];
+    }
+    expressApp.use(context, httpProxyMiddleware(proxy));

```

Comment:
Sorry for being so nitpicky, but can please only register the middleware inside each `if` condition?

Right now if for whatever reason the config is neither a string nor an object (even though the schema advises against it) we register a proxy middleware with an empty config. For convenience, and to help TypeScript, you can put `httpProxyConfig[context]` in a constant before checking its type.

This is what I propose:

```ts
    const config = httpProxyConfig[context];
    if (typeof config === 'string') {
      expressApp.use(context, { target: config });
    } else if (typeof config === 'object') {
      expressApp.use(context, config);
    }
```

Thanks!

Reply:
Okay, no worries, I see what you are saying.